packages/edit-site/src/store/actions.js

```diff
@@ -175,6 +175,21 @@ export function setTemplatePart( templatePartId ) {
 	};
 }
 
+/**
+ * Action that sets a navigation menu.
+ *
+ * @param {string} navigationMenuId The Navigation Menu Post ID.
+ *
+ * @return {Object} Action object.
+ */
+export function setNavigationMenu( navigationMenuId ) {
+	return {
+		type: 'SET_EDITED_POST',
+		postType: 'wp_navigation',
+		id: navigationMenuId,
+	};
+}
```

Comment:
Notice how this sets generic state for the edited post type. There is a corresponding selector `getEditedPostType` which retrieves this information. Currently this is heavily biased towards `templates` / `template_parts` and we will need a refactoring exercise to decouple all this to allow us to work with other post types.